sdk/clientcore/annotation-processor-test/src/main/java/io/clientcore/annotation/processor/test/ParameterizedMultipleHostServiceImpl.java

```diff
@@ -57,9 +57,16 @@ public HttpBinJSON get(String scheme, String hostPart1, String hostPart2) {
         int responseCode = networkResponse.getStatusCode();
         boolean expectedResponse = responseCode == 200;
         if (!expectedResponse) {
-            String errorMessage = networkResponse.getValue().toString();
-            networkResponse.close();
-            throw new HttpResponseException(errorMessage, networkResponse, null);
+            BinaryData value = networkResponse.getValue();
+            if (value == null || value.toBytes().length == 0) {
+                networkResponse.close();
+                throw CoreUtils.instantiateUnexpectedException(responseCode, networkResponse, null, null);
+            } else {
+                ParameterizedType returnType = CoreUtils.createParameterizedType(io.clientcore.annotation.processor.test.implementation.models.HttpBinJSON.class);
+                Object decoded = CoreUtils.decodeNetworkResponse(value, jsonSerializer, returnType);
+                networkResponse.close();
+                throw CoreUtils.instantiateUnexpectedException(responseCode, networkResponse, value, decoded);
+            }
```

Comment:
Would we be better off creating a utility method in Core to handle this logic, rather than repeating in every generated API?